src/main/java/io/vertx/ext/stomp/StompClientConnection.java

```diff
@@ -0,0 +1,548 @@
+   *                       ({@code COMMIT}).
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection commit(String id, Map<String, String> headers, Handler<Frame> receiptHandler);
+
+  /**
+   * Aborts a transaction.
+   *
+   * @param id the transaction id, must not be {@code null}
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection abort(String id);
+
+  /**
+   * Aborts a transaction.
+   *
+   * @param id             the transaction id, must not be {@code null}
+   * @param receiptHandler the handler invoked when the {@code RECEIPT} frame associated with the
+   *                       transaction cancellation has been processed by the server. The handler receives the sent
+   *                       frame ({@code ABORT}).
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection abort(String id, Handler<Frame> receiptHandler);
+
+  /**
+   * Aborts a transaction.
+   *
+   * @param id      the transaction id, must not be {@code null}
+   * @param headers additional headers to send to the server. The {@code transaction} header is replaced by the
+   *                value passed in the @{code id} parameter
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection abort(String id, Map<String, String> headers);
+
+  /**
+   * Aborts a transaction.
+   *
+   * @param id             the transaction id, must not be {@code null}
+   * @param headers        additional headers to send to the server. The {@code transaction} header is replaced by the
+   *                       value passed in the @{code id} parameter
+   * @param receiptHandler the handler invoked when the {@code RECEIPT} frame associated with the
+   *                       transaction cancellation has been processed by the server. The handler receives the sent
+   *                       frame ({@code ABORT}).
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection abort(String id, Map<String, String> headers, Handler<Frame> receiptHandler);
+
+  /**
+   * Disconnects the client. Unlike the {@link #close()} method, this method send the {@code DISCONNECT} frame to the
+   * server.
+   *
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection disconnect();
+
+  /**
+   * Disconnects the client. Unlike the {@link #close()} method, this method send the {@code DISCONNECT} frame to the
+   * server.
+   *
+   * @param receiptHandler the handler invoked when the {@code RECEIPT} frame associated with the
+   *                       disconnection has been processed by the server. The handler receives the sent
+   *                       frame ({@code DISCONNECT}).
+   * @return the current {@link StompClientConnection}
+   */
+  @Fluent
+  StompClientConnection disconnect(Handler<Frame> receiptHandler);
+
+
```

Comment:
I think it would also be good to let the user send Frame objects that they created using Frames class.